src/App.tsx

```diff
@@ -1,73 +1,93 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import './App.scss';
+import cn from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { DropdownMenu } from './components/DropdownMenu';
+
+
+function debounce(callback: Function, delay: number = 300) {
+  let timerId: number = 0;
+  
+  return (...args: any) => {
+    window.clearTimeout(timerId);
+    timerId = window.setTimeout(() => {
+      callback(...args)
+    }, delay)
+  };
+}
+
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [name, setName] = useState('');
+  const [appliedPerson, setAppliedPerson] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const applyPerson = debounce(setAppliedPerson, 1000);
+
+  const filteredPeople = useMemo(() => {
+    return peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(appliedPerson.toLowerCase()),
+    );
+  }, [appliedPerson]);
+
+  const onSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    applyPerson(e.target.value);
+    setSelectedPerson(null);
+  };
+
+  const onSelected = (person: Person) => () => {
+    setName(person.name);
+    setAppliedPerson('');
```

Comment:
Clearing `appliedPerson` here might lead to unexpected behavior if the user expects the input field to retain the selected person's name. Consider whether this is the intended behavior.